x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportGetInferenceModelAction.java

```diff
@@ -53,8 +63,20 @@ protected void doExecute(
         GetInferenceModelAction.Request request,
         ActionListener<GetInferenceModelAction.Response> listener
     ) {
-        modelRegistry.getUnparsedModelMap(request.getModelId(), ActionListener.wrap(modelConfigMap -> {
-            var unparsedModel = UnparsedModel.unparsedModelFromMap(modelConfigMap.config(), modelConfigMap.secrets());
+        boolean modelIdIsWildCard = Strings.isAllOrWildcard(request.getModelId());
+
+        if (request.getTaskType() == TaskType.ANY && modelIdIsWildCard) {
+            getAllModels(listener);
+        } else if (modelIdIsWildCard) {
+            getModelsByTaskType(request.getTaskType(), listener);
+        } else {
+            getSingleModel(request.getModelId(), listener);
```

Comment:
If I understand this logic correctly, the endpoint
```
/_inference/WHATEVER-TASKTYPE/modelId
```
always returns the model with the modelId, no matter whether the task type is correct for that model.